api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1248,9 +1248,72 @@ const (
 // +kubebuilder:validation:Enum=Karpenter
 type Provisioner string
 
+// Configures when and how to scale down cluster nodes.
+type ScaleDownConfig struct {
+	// enabled determines whether the Cluster Autoscaler should scaled down this cluster.
+	Enabled bool `json:"enabled"`
```

Comment:
Have you considered using some kind of enum to represent the state instead of a bool? Bools don't typically evolve well over time, we prefer enums with meaningful values that describe the behaviour that they configure

Reply:
I did. But I wanted to keep the API matching https://github.com/openshift/cluster-autoscaler-operator/blob/master/pkg/apis/autoscaling/v1/clusterautoscaler_types.go#L164-L164

Reply:
I'm leaning towards consistency with the existing cao API and the abstracted flag here as well.